docs/index.md

```diff
@@ -1,94 +1,217 @@
+
+- [Assimp](www.assimp.org) Assimp provides facilities to load mesh files of various file formats
+
+- [dear IMGUI](https://github.com/ocornut/imgui) dear IMGUI is an easy to use immediate mode graphics user interface library, its widely used in the industry and we will be using it to display interactive options
+
+#### Project Layout
+
+- `.github` contains github related files to enable this project's existence
+- `docs` contains this very documentation describing each chapter
+- `lib` contains all external/third party libraries like `GLFW` or `AssImp`
+- `src` contains all the example source code behind this tutorial series
+
+# 0. Initial Setup
+
+```todo
+either clone repo, open empty solution with project and empty main.cpp
+or explain how to create a project in VS, set it up so that it makes sense,
+explain include/lib folders, how to set output folder and why Win32 is not
+an option really anymore either
+```
+
+Assume we cloned the empty main, already setup with GLFW
+
+Ok let's start with the whole example for a HelloWindow first
+
+```cpp
+#include <GLFW/glfw3.h>
+#include <cstdint>
+#include <iostream>
+
+int main(int argc, char* argv[])
+{
+    if (!glfwInit())
+    {
+        std::cout << "GLFW: Unable to initialize\n";
+        return -1;
+    }
+
+    const GLFWmonitor* primaryMonitor = glfwGetPrimaryMonitor();
+    const GLFWvidMode* videoMode = glfwGetVideoMode(primaryMonitor);
+    const int32_t width = static_cast<int32_t>(videoMode->width * 0.9f);
+    const int32_t height = static_cast<int32_t>(videoMode->height * 0.9f);
+
+    glfwWindowHint(GLFW_SCALE_TO_MONITOR, GLFW_FALSE);
+    glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
+    const GLFWwindow* window = glfwCreateWindow(
+        width,
+        height,
+        "LearnD3D11 - Hello Window",
+        nullptr,
+        nullptr);
+    if (window == nullptr)
+    {
+        std::cout << "GLFW: Unable to create window\n";
+        glfwTerminate();
+        return -1;
+    }
+
+    const int32_t windowLeft = videoMode->width / 2 - width / 2;
+    const int32_t windowTop = videoMode->height / 2 - height / 2;
+    glfwSetWindowPos(window, windowLeft, windowTop);
+
+    while (!glfwWindowShouldClose(window))
     {
-        public SomeCSharpCodeHere()
-        {
-        }
+        glfwPollEvents();
+        // future update code
+        // future render code
     }
-    ```
 
-    ```xml
-    <Why Not="Some">
-        <Xml/>
-    </Why>
-    ```
+    glfwDestroyWindow(window);
+    glfwTerminate();
+    return 0;
+}
+```
+
+Now let me explain what all these things mean.
+
+```cpp
+#include <GLFW/glfw3.h>
+```
+
+C++ needs to know where all the definitions and declarations are coming from.
+
+```cpp
+    if (!glfwInit())
+    {
+        std::cout << "GLFW: Unable to initialize\n";
+        return -1;
+    }
+```
+
+Pretty obvious, right? Try to initialize `GLFW`. If it fails to do so, let the user know and end the program, since there is no point in going further.
```

Comment:
recommended phrase change to: 
"Pretty obvious, right? `glfwInit()` tries to initialize `GLFW`. If it fails to do so, let the user know and end the program, since there is no point in going further."
 
This is recommended because when reading the phrase the first time around, it can be very convoluted for what 
Pretty obvious, right? Try to initialize `GLFW`." is referring to; does it mean inside glfwInit, was something missed? - the small amount of extra context clears up any confusion with the statment for what is pretty obvious. You also could just say: "The following code is pretty obvious, right? ..." <-- this has the same effect but keeps the light hearted tone

Reply:
can you provide the actual markdown aleady? then i dont have to pick it out myself from your suggestion

Reply:
sure, I will do it after - just need to sort through everything first (plus I also need to justify my suggestion - else its crass of me to suggest something arbitrarily 'just because'

Reply:
Mark down:  
Pretty obvious, right? `glfwInit()` tries to initialize `GLFW`. If it fails to do so, let the user know and end the program, since there is no point in going further."

Reply:
> Mark down: `` Pretty obvious, right? `glfwInit()` tries to initialize `GLFW`. If it fails to do so, let the user know and end the program, since there is no point in going further." ``

If i am not blind, thats exactly what I have written :) so not sure what you want me to change now

Reply:
🤷

Reply:
I was asking for the exact markdown you want me the thing to change, but you provided my initial text in the markdown, not sure if that was your intention?!?

Reply:
my change is minor, it just adds the `glfwInit()` section for clarity - i'm sorry for being very unclear

Reply:
OK I also found the text you were referring to :)